### DISCUSSION TITLE
Downstream analysis tools pVACbind

### DISCUSSION THREAD

**Discussion by martinez14recerca** [griffithlab/pVACtools#1156] [General]
Hello,
I have been using pVACbind to select candidate neoantigens for class I and II. For every single patient, I have worked with the "all.epitopes.tsv" using the Median IC50 Score to determine weak binders and strong binders. From this results, I have generated a second .tsv containing the following information: Chromosome, Start, Stop, Reference, Variant, Transcript, HLA Allele, Peptide length, Epitope Sequence (as shown in the document attached. Part of the information is given by pVACbind and the rest is from previous analysis that I have). I have generated single .tsv for each patient (without adding the patient and WB_SB columns). 

Now my intention is to run the downstream analysis tools (NetChop predict cleavages and NetMHCStab predict stability). Since I am not using the direct "all.epitopes.tsv", I run the following command in my computer: 
`pvacbind run \ /home/immunocart/ANALISISTCRDOWNSTREAM/LLC_001_I.tsv \ LLC_001_I \ HLA-A*11:01,HLA-A*24:02,HLA-C*16:01 \ all \ /home/immunocart/ANALISISTCRDOWNSTREAM/RESULTS/LLC_001_I_output \ --net-chop-method cterm \ --netmhc-stab \ -e1 8,9,10,11`
It seems the code is running but at the end I am not getting any file: 
`Executing MHC Class I predictions Splitting FASTA into smaller chunks Splitting FASTA into smaller chunks - Entries 1-2 Completed Fasta file /home/immunocart/ANALISISTCRDOWNSTREAM/RESULTS/LLC_001_I_output/MHC_Class_I/tmp/LLC_001_I.8.fa.split_1-2 is empty. Skipping Splitting FASTA into smaller chunks Splitting FASTA into smaller chunks - Entries 1-2 Completed Fasta file /home/immunocart/ANALISISTCRDOWNSTREAM/RESULTS/LLC_001_I_output/MHC_Class_I/tmp/LLC_001_I.9.fa.split_1-2 is empty. Skipping Splitting FASTA into smaller chunks Splitting FASTA into smaller chunks - Entries 1-2 Completed Fasta file /home/immunocart/ANALISISTCRDOWNSTREAM/RESULTS/LLC_001_I_output/MHC_Class_I/tmp/LLC_001_I.10.fa.split_1-2 is empty. Skipping Splitting FASTA into smaller chunks Splitting FASTA into smaller chunks - Entries 1-2 Completed Fasta file /home/immunocart/ANALISISTCRDOWNSTREAM/RESULTS/LLC_001_I_output/MHC_Class_I/tmp/LLC_001_I.11.fa.split_1-2 is empty. Skipping No output files were created. Aborting. No MHC class II alleles chosen. Skipping MHC class II predictions.`

I would like to ask if you could detect the error or propose another alternative, since I thought the input file contains all the necessary information for the programs to run. Thanks in advance!

[GitHub_20241031.xlsx](https://github.com/user-attachments/files/17587273/GitHub_20241031.xlsx)

**Comment by susannasiebert** [comment]
I don't think rerunning pvacbind run is the correct approach here. pVACbind expects a FASTA file, not a TSV. Please have a look at the [optional downstream analysis tools](https://pvactools.readthedocs.io/en/latest/pvacbind/optional_downstream_analysis_tools.html) section of the documentation particularly the two commands to run NetChop and NetMHCstabpan standalone. As long as your TSV file contains the `Mutation` and `Epitope Seq` columns, this should work. These two columns correspond to the FASTA header of the original input fasta entry the epitope came from as well as the epitope, respectively.
